codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Codec.java

```diff
@@ -40,8 +40,11 @@ Http2FrameCodec frameCodec() {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        ctx.pipeline().addBefore(ctx.executor(), ctx.name(), null, frameCodec);
-        ctx.pipeline().addBefore(ctx.executor(), ctx.name(), null, multiplexCodec);
+        // We use addAfter(...) because if we would use addBefore(...) we could end up with an incorrect
+        // order of handlers in the pipeline when replace(...) is used with the Http2Codec.
+        // See https://github.com/netty/netty/issues/6881.
+        ctx.pipeline().addAfter(ctx.executor(), ctx.name(), null, multiplexCodec);
```

Comment:
nope we want to use the same executor as the one that was used by the `Http2Codec`, just like we did before.